### ISSUE TITLE
Generic/InlineControlStructures: sniff does not allow while without body

### ISSUE THREAD

**Issue by jrfnl** [squizlabs/PHP_CodeSniffer#2822]
A `while` control structure without body is perfectly valid PHP but gets flagged by the `Generic.ControlStructures.InlineControlStructures` sniff.

```
$i = 10;
while ($i > 0 && --$i);

// $i is now 0
```
See: https://3v4l.org/RXNqL

As far as I can see, the sniff is intended to find inline control structures *with* a body, but *without* braces and fix those. An inline control structure without a body should be outside the scope of this sniff.

I've got a fix ready for this, but found that a number of existing unit test would start failing (line 42, 43, 226, 228 and 230).

@gsherwood With that in mind, I'd like second opinion on how the sniff should handle `while` structures without body.

As an alternative to the fix I've got prepared, I could create a fix which would maintain the existing behaviour, but would give `while` structures without a body a different error code.

Opinions ?

**Comment by gsherwood** [comment]
> As far as I can see, the sniff is intended to find inline control structures _with_ a body, but _without_ braces and fix those. An inline control structure without a body should be outside the scope of this sniff.

100% agree.

> With that in mind, I'd like second opinion on how the sniff should handle `while` structures without body.

I think the sniff should ignore them. Happy to change the tests to align with this change.

**Comment by jrfnl** [comment]
@gsherwood Thanks for he response. PR #2827 should fix it in that case.

**Comment by jrfnl** [comment]
Just realized that `for` statements can also exist without body. I've updated the PR to include a fix for that too.

```php
for ($i = 1, $j = 0; $i <= 10; $j += $i, print $i, $i++);
```
